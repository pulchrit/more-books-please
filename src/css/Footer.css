.footer {
    border-top: 1px solid rgba(10, 34, 73, 0.8);
    width: 100%;
    padding: 10px;
    margin-top: 20px;
    text-align: center;
}

.footer-copy {
    margin: 4px;
}

.footer-copy a {
    text-decoration: underline;
    font-size: .9em;
} 

a:link {
    color: #0a2249;
}
  
a:visited {
    color: #490A30;
}
  
a:hover, a:focus {
    color: #30490A;
} 
  
a:active {
    text-decoration-style: dotted;
}