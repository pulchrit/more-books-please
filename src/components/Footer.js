import React from 'react';
import '../css/Footer.css';

const Footer = () => {
    return (
        <footer className="footer">
            <p className="footer-copy"><a href="https://developer.nytimes.com">Data provided by The New York Times</a></p>
            <p className="footer-copy"><a className="main-a-style" href="https://openlibrary.org/">Data provided by OpenLibrary</a></p>
            <p className="footer-copy"> &copy;2019 <a className="main-a-style" href="mailto:pulchrit@gmail.com">Melissa Lafranchise</a></p>
        </footer>
    );
}

export default Footer;