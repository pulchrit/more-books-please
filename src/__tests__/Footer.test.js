import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Footer from '../components/Footer';

describe ("Footer", () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Footer /> 
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
            <Footer />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

})


