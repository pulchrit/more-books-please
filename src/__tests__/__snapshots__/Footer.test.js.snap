// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders UI as expected 1`] = `
<footer
  className="footer"
>
  <p
    className="footer-copy"
  >
    <a
      href="https://developer.nytimes.com"
    >
      Data provided by The New York Times
    </a>
  </p>
  <p
    className="footer-copy"
  >
    <a
      className="main-a-style"
      href="https://openlibrary.org/"
    >
      Data provided by OpenLibrary
    </a>
  </p>
  <p
    className="footer-copy"
  >
     Â©2019 
    <a
      className="main-a-style"
      href="mailto:pulchrit@gmail.com"
    >
      Melissa Lafranchise
    </a>
  </p>
</footer>
`;
